test/jdk/javax/swing/JRadioButton/bug4380543.java

```diff
@@ -0,0 +1,160 @@
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+/* @test
+ * @bug 4380543
+ * @key headful
+ * @requires (os.family == "windows")
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @summary setMargin() does not work for AbstractButton
+ * @run main/manual bug4380543
+*/
+
+import java.awt.BorderLayout;
+import java.awt.Color;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.awt.Insets;
+
+import javax.swing.BoxLayout;
+import javax.swing.JButton;
+import javax.swing.JCheckBox;
+import javax.swing.JFrame;
+import javax.swing.JPanel;
+import javax.swing.JRadioButton;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+import javax.swing.UnsupportedLookAndFeelException;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class bug4380543 {
+    static testFrame testObj;
+    static String instructions
+            = "INSTRUCTIONS:" +
+            "\n 1. This is a Windows specific test. If you are not on " +
+            "Windows, press Pass." +
+            "\n 2. Check if the Left insets(margins) is set visually " +
+            "similar to other three sides around Radio Button and CheckBox" +
+            "(insets set to 20 on all 4 sides)." +
+            "\n 3. If Left insets(margins) appear Empty, press Fail, " +
+            "else press Pass.";
+    static PassFailJFrame passFailJFrame;
+
+    public static void main(String args[]) throws Exception {
+
+        SwingUtilities.invokeAndWait(new Runnable() {
+            public void run() {
+                try {
+                    passFailJFrame = new PassFailJFrame(instructions);
+                    testObj = new testFrame();
+                    //Adding the Test Frame to handle dispose
+                    PassFailJFrame.addTestFrame(testObj);
+                    PassFailJFrame.positionTestFrame(testObj, PassFailJFrame.Position.HORIZONTAL);
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
+            }
+        });
+        passFailJFrame.awaitAndCheck();
+    }
+}
+
+class testFrame extends JFrame implements ActionListener {
+    JPanel buttonsPanel;
+
+    Map<String, String> lookAndFeelMaps = new HashMap<String, String>();
+    public testFrame() throws InterruptedException {
+        initMap();
+        initComponents();
+    }
+
+    public void initMap()
+    {
+        String sLnF;
+        String sMapKey;
+        UIManager.LookAndFeelInfo[] lookAndFeel = UIManager.getInstalledLookAndFeels();
+        for (UIManager.LookAndFeelInfo look : lookAndFeel) {
+
+            sLnF = look.getClassName();
+            sMapKey = sLnF.substring(sLnF.lastIndexOf(".")+1);
+            sMapKey = sMapKey.replaceAll("LookAndFeel","");
+            sMapKey = sMapKey.trim();
+
+            lookAndFeelMaps.put(sMapKey, sLnF);
+        }
```

Comment:
```suggestion
        for (UIManager.LookAndFeelInfo look : lookAndFeel) {
            lookAndFeelMaps.put(look.getName(), look.getClassName());
        }
```

I guess I mentioned it earlier. I think this way is cleaner and more user-friendly: the user will see the name of the Look and Feel rather than part of its class name.

The local variables `sLnF` and `sMapKey` declared above the look become unnecessary in this case.

If you use `LinkedHashMap`, the order of the L&Fs will be preserved the map will be iterated for creating the buttons.

Reply:
The reason for doing this was to use the same key for button creation (name of button) and then to link that with the button action, so changing this will affect other logics.